File: Competitive Programming 3/Section 1.4/Game (Chess)/278 - Chess/Python/main.py
import math

def main():
    t = int(input())
    for _ in range(t):
        typ,n,m = [int(c) if i > 0 else c for i,c in enumerate(input().split())]
        maxAns=0
        if typ == 'r':
            maxAns = min(m,n)
        elif typ == 'k':
            if n == 1 or m == 1:
                maxAns = max(n,m)
            elif n==2 or m == 2:
                l = max(n,m)
                localAns = 0
                localAns = (l+1)//4
                localAns*=2
                if l%4 == 1 or l%4 == 2:
                    localAns += l%4
                maxAns = localAns * 2
            else:
                maxAns = (n*m+1)//2
        elif typ == 'Q':
            maxAns = n-1 if m == n and (n==2 or m == 3) else min(m,n)
        else:
            maxAns = math.ceil(m/2)*math.ceil(n/2)
        print(maxAns)
main()

File: Competitive Programming 3/Section 1.4/Game (Chess)/278 - Chess/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in);
    public static void main(String[]args){
        char type;
        for(int t = s.nextInt();t>0;t--){
            type = s.next().charAt(0);
            int n=s.nextInt(),m=s.nextInt(),maxAns=0;
            if(type=='r'){
                maxAns=Math.min(m,n);
            }else if(type=='k'){
                if(n==1||m==1){
                    maxAns = Math.max(n,m);
                }else if (n==2 || m ==2){
                    int l = Math.max(n,m);
                    int localAns = 0;
                    localAns = (l+1)/4;
                    localAns*=2;
                    if(l%4==1 || l%4==2){
                        localAns += l%4;
                    }
                    maxAns = localAns*2;
                }else{
                    maxAns = (n*m+1)/2;
                }
            }else if(type=='Q'){
                maxAns = m == n && (n==2||m==3) ? n-1 : Math.min(m,n);
            }else{
                maxAns = (int)Math.ceil((float)m/2)*(int)Math.ceil((float)n/2);
            }
            System.out.printf("%d\n",maxAns);
        }
    }
}